### PR TITLE
Release v3.0.3

### PR DESCRIPTION
**Pull request by CyberDem0n**
- update release notes
- bump Patroni version
- bump pyright version and resolve issues
- add readthedocs config file

### CODE REVIEW

**Comment by CyberDem0n** [comment]
It was affecting only unit-tests, the actual code didn't had any problems.

**Comment by barthisrael** [comment]
Should we reference [this PR](https://github.com/zalando/patroni/pull/2639) as a bugfix in the release notes?

**Comment by CyberDem0n** [comment]
> Should we reference https://github.com/zalando/patroni/pull/2639 as a bugfix in the release notes?

I don't even know if someone even use this feature, it is very basic comparing to using normal config files.
But, if you like please make a suggestion in the code :)

**Comment by barthisrael** [comment]
> > Should we reference #2639 as a bugfix in the release notes?
> 
> I don't even know if someone even use this feature, it is very basic comparing to using normal config files. But, if you like please make a suggestion in the code :)

Thanks for your thoughts. Yeah, I don't know either. I personally just hit that issue by coincidence while testing/writing docstrings. I don't think I would have hit that in a day-to-day basis. But I think it wouldn't hurt to add a reference just in case. Let me suggest something then

File: docs/releases.rst [hunk 1]
```diff
@@ -3,6 +3,56 @@
...
+  The ``\n`` symbol was rendered as it is, instead of the actual newline symbol.
+
+
+**Bugfixes**
+
+- Fixed issue in Citus support (Alexander)
+
+  If the REST API call from the promoted worker to the coordinator failed during switchover it was leaving the given Citus group blocked during indefinite time.
+
+
```
Comment (barthisrael) [inline docs/releases.rst:55]: - Allow `etcd3` URL in `--dcs-url` option of `patronictl`

  If users attempted to pass a `etcd3` URL through `--dcs-url` option of `patronictl` they would face an exception.

File: docs/releases.rst [hunk 2]
```diff
@@ -3,6 +3,56 @@
...
+- Call ``initdb`` directly when initializing a new cluster (Matt Baker)
+
+  Previously it was called via ``pg_ctl``, what required a special quoting of parameters passed to ``initdb``.
+
+- Added before stop hook (Le Duane)
+
+  The hook could be configured via ``postgresql.before_stop`` and is executed right before ``pg_ctl stop``. The exit code doesn't impact shutdown process.
+
+- Added support for custom Postgres binary names (Israel, Polina Bungina)
```
Comment (martinmarques) [inline docs/releases.rst:31]: We should put full names

File: docs/releases.rst [hunk 3]
```diff
@@ -3,6 +3,56 @@
...
+
+  When using a custom Postgres distribution it may be the case that the Postgres binaries are compiled with different names other than the ones used by the community Postgres distribution. Custom binary names could be configured using ``postgresql.bin_name.*`` and ``PATRONI_POSTGRESQL_BIN_*`` environment variables.
+
+
+**Improvements**
+
+- Various improvements of ``patroni --validate-config`` (Polina)
```
Comment (martinmarques) [inline docs/releases.rst:38]: Idem before

File: docs/releases.rst [hunk 4]
```diff
@@ -3,6 +3,56 @@
...
+
+  -  Make ``bootstrap.initdb`` optional. It is only required for new clusters, but ``patroni --validate-config`` was complaining if it was missing in the config.
+  -  Don't error out when ``postgresql.bin_dir`` is empty or not set. Try to first find Postgres binaries in the default PATH instead.
+  -  Make ``postgresql.authentication.rewind`` section optional. If it is missing, Patroni is using the superuser.
+
+- Improved error reporting in ``patronictl`` (Israel)
```
Comment (martinmarques) [inline docs/releases.rst:44]: Idem before

File: docs/releases.rst [hunk 5]
```diff
@@ -3,6 +3,56 @@
...
+
+  The ``\n`` symbol was rendered as it is, instead of the actual newline symbol.
+
+
+**Bugfixes**
+
+- Fixed issue in Citus support (Alexander)
```
Comment (martinmarques) [inline docs/releases.rst:51]: Idem before

File: docs/releases.rst (hunk 1 above)
Comment (martinmarques) [reply docs/releases.rst:55]: Use full names for the contributors

**Review by martinmarques — CHANGES_REQUESTED** [review]
Uniform mention of committers in the Release Notes

File: docs/releases.rst (hunk 2 above)
Comment (hughcapet) [reply docs/releases.rst:31]: We actually only put full names at the first mention. Is that a problem?

Comment (martinmarques) [reply docs/releases.rst:31]: The problem I see is consistency overall. I read "Polina Bungina" in one place and "Polina" later in the Release Notes and that would at least raise an eye-brow.
Is there a reason not to have the full name on every feature/bug fix? The Postgres release notes follow the same convention, which I've always found pretty nice overall (everything in the Postgres docs is very well written and organized).

Comment (hughcapet) [reply docs/releases.rst:31]: I see and ultimately agree. I will change it in a bit, hope @CyberDem0n doesn't mind.

**Comment by CyberDem0n** [comment]
Well...
If we start doing that than the whole file should be updated :)
But IMO, it shouldn't be done in this PR.

**Comment by hughcapet** [comment]
> But IMO, it shouldn't be done in this PR.

Exactly. This is what I wanted to do next

**Comment by CyberDem0n** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Pull request merged**